src/boot/efi/util.c

```diff
@@ -501,3 +501,17 @@ EFI_STATUS log_oom(void) {
         log_error_stall(L"Out of memory.");
         return EFI_OUT_OF_RESOURCES;
 }
+
+VOID *memmem_safe(const VOID *haystack, UINTN haystack_len, const VOID *needle, UINTN needle_len) {
+        assert(haystack);
+        assert(needle);
```

Comment:
i think it should be OK for generic functions like this to accept NULL for both args as long as the length is also specified as 0. i.e. change these asserts to:

```c
assert(haystack || haystack_len == 0);
assert(needle || needle_len == 0);
```